std/algorithm/searching.d

```diff
@@ -4027,6 +4101,18 @@ template skipOver(alias pred = "a == b")
     assert(s4.skipOver!isWhite && s3.empty);
 }
 
+/// Variadic skipOver
```

Comment:
These can probably be nothrow pure too?

Reply:
Due to UTF auto-decoding strings aren't `nothrow` by default. It be prevented by adding a few `byCodeUnit` in there, but typically one test is added to ensure that all attributes work (i.e. `@safe @nogc pure nothrow` in this case), s.t. the other tests don't need to bother with `static immutable`, `byCodeUnit` or other unrelated "tricks". This is especially important because this test is the publicly documented one.

The test with all attributes is a few lines below.

Reply:
But can they still be marked pure without changing the internals? I see nothing that should prevent it.

If so IMO it will add to the documentation, as the user can see it works in pure code.

Reply:
Unittest attributes aren't displayed on the documentation, but yes `pure` could be added here.